nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BetaInc.java

```diff
@@ -0,0 +1,50 @@
+package org.nd4j.linalg.api.ops.custom;
+
+import lombok.NonNull;
+import org.nd4j.autodiff.samediff.SDVariable;
+import org.nd4j.autodiff.samediff.SameDiff;
+import org.nd4j.base.Preconditions;
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.api.ops.DynamicCustomOp;
+
+import java.util.Collections;
+import java.util.List;
+
+public class BetaInc extends DynamicCustomOp {
+
+    public BetaInc() {}
+
+    public BetaInc(@NonNull INDArray a_input, @NonNull INDArray b_input, @NonNull INDArray x_input,
+                   INDArray output) {
+        addInputArgument(a_input, b_input, x_input);
+        addOutputArgument(output);
```

Comment:
Null check on the output argument - we don't want to call `addOutputArgument(null)`

Reply:
So we need ability to pass output == null, but don't call addOutputArguments(), right? So we need if (output != null) and @NonNull will not work.

Reply:
Correct. No `@NonNull` annotation on the output arg, and then just:
```
if(output != null){
    addOutputArgument(output);
}
```

Reply:
Done.